SH4RE/SH4RE/CreateListing/CreateListingView.swift

```diff
@@ -362,8 +358,36 @@ struct CreateListingView: View {
             title = editListing.title
             description = editListing.description
             cost = editListing.price
-            postalCode = editListing.address["postalCode"] as? String ?? "ERROR"
+            costText = String(format:"%.2f", editListing.price)
+            postalCode = editListing.postalCode
+
             availabilityCalendar.selectedDates = editListing.availability
+            Firestore.firestore().collection("Listings").document(editListing.id).collection("Requests").getDocuments() {(snapshot, error) in
+                snapshot?.documents.forEach({ (document) in
+                    let data = document.data()
+                    let start = data["start"] as? Timestamp ?? Timestamp(date:Date(timeIntervalSince1970: 0))
+                    let end = data["end"] as? Timestamp ?? Timestamp(date:Date(timeIntervalSince1970: 0))
+                    let status = data["status"] as? String ?? ""
+                    
+                    if(status == "accepted" || status == "pending"){
+                        var booking:[Date] = []
+                        if(end.dateValue() != Date(timeIntervalSince1970: 0)){
+                            var date = start.dateValue()
+                            let fmt = DateFormatter()
+                            fmt.dateFormat = "dd/MM/yyyy"
+                            while date <= end.dateValue() {
+                                fmt.string(from: date)
+                                booking.append(date)
+                                date = Calendar.current.date(byAdding: .day, value: 1, to: date)!
+                            }
+                        }else{
+                            booking.append(start.dateValue())
+                        }
+                        availabilityCalendar.disabledDates.append(contentsOf: booking)
+                    }
+                })
+            }
+            
```

Comment:
this code disables calendar for dates the listing is booked so owner cannot overwrite